src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenRefReadonlyParameterTests.cs

```diff
@@ -0,0 +1,474 @@
+                		)
+                	// Method begins at RVA 0x2067
+                	// Code size 1 (0x1)
+                	.maxstack 8
+                	IL_0000: ret
+                } // end of method C::M
+                .method public hidebysig specialname rtspecialname 
+                	instance void .ctor () cil managed 
+                {
+                	// Method begins at RVA 0x2069
+                	// Code size 7 (0x7)
+                	.maxstack 8
+                	IL_0000: ldarg.0
+                	IL_0001: call instance void [netstandard]System.Object::.ctor()
+                	IL_0006: ret
+                } // end of method C::.ctor
+            } // end of class C
+            """);
+
+        static void verify(ModuleSymbol m)
+        {
+            var p = m.GlobalNamespace.GetMember<MethodSymbol>("C.M").Parameters.Single();
+            Assert.Equal(RefKind.RefReadOnlyParameter, p.RefKind);
+        }
+    }
+
+    [Fact]
+    public void Method_Virtual()
+    {
+        var source = """
+            class C
+            {
+                public virtual void M(ref readonly int p) { }
+            }
+            """;
+        var verifier = CompileAndVerify(source, targetFramework: TargetFramework.NetStandard20,
+            sourceSymbolValidator: verify, symbolValidator: verify);
+        verifier.VerifyDiagnostics();
+        verifier.VerifyTypeIL("C", """
+            .class private auto ansi beforefieldinit C
+                extends [netstandard]System.Object
+            {
+                // Methods
+                .method public hidebysig newslot virtual 
+                	instance void M (
+                		[in] int32& modreq([netstandard]System.Runtime.InteropServices.InAttribute) p
+                	) cil managed 
+                {
+                	.param [1]
+                		.custom instance void System.Runtime.CompilerServices.RequiresLocationAttribute::.ctor() = (
+                			01 00 00 00
+                		)
+                	// Method begins at RVA 0x2067
+                	// Code size 1 (0x1)
+                	.maxstack 8
+                	IL_0000: ret
+                } // end of method C::M
+                .method public hidebysig specialname rtspecialname 
+                	instance void .ctor () cil managed 
+                {
+                	// Method begins at RVA 0x2069
+                	// Code size 7 (0x7)
+                	.maxstack 8
+                	IL_0000: ldarg.0
+                	IL_0001: call instance void [netstandard]System.Object::.ctor()
+                	IL_0006: ret
+                } // end of method C::.ctor
+            } // end of class C
+            """);
+
+        static void verify(ModuleSymbol m)
+        {
+            var p = m.GlobalNamespace.GetMember<MethodSymbol>("C.M").Parameters.Single();
+            Assert.Equal(RefKind.RefReadOnlyParameter, p.RefKind);
+        }
+    }
+
+    [Fact]
+    public void Constructor()
```

Comment:
Consider adding a test with primary constructor or record if we don't have some already. I assume both are parsing errors.

Reply:
Added tests, no parsing errors.